packages/web/src/components/create-playlist/PlaylistForm.tsx

```diff
@@ -0,0 +1,258 @@
+          type={ButtonType.SECONDARY}
+          onClick={debounce(onSave, 500, { leading: true })}
+        />
+      </div>
+    </div>
+  )
+}
+
+const CreatePlaylistActions = ({ onSave }: { onSave: () => void }) => {
+  return (
+    <div className={styles.createPlaylistActionsContainer}>
+      <Button
+        rightIcon={<IconCheck />}
+        text={messages.createPlaylistButtonText}
+        type={ButtonType.PRIMARY}
+        onClick={debounce(onSave, 500, { leading: true })}
+      ></Button>
+    </div>
+  )
+}
+
+const PlaylistForm = ({
+  isAlbum = false,
+  metadata,
+  onSave: onSaveParent,
+  onCancel,
+  onDelete,
+  onOpenArtworkPopup,
+  onCloseArtworkPopup,
+  isEditMode = false
+}: PlaylistFormProps) => {
+  const [formFields, setFormFields] = useState<PlaylistFormFields>({
+    artwork: {},
+    ...schemas.newCollectionMetadata(metadata)
+  })
+  const [errors, setErrors] = useState({
+    playlistName: false,
+    artwork: false
+  })
+
+  const coverArt = useCollectionCoverArt(
+    formFields.playlist_id,
+    formFields?._cover_art_sizes ? formFields._cover_art_sizes : null,
+    SquareSizes.SIZE_1000_BY_1000
+  )
+
+  // On receiving new, defined metadata, update the form fields
+  useEffect(() => {
+    if (metadata) {
+      setFormFields(oldFormFields => ({
+        ...schemas.newCollectionMetadata(metadata),
+        artwork: oldFormFields.artwork,
+        playlist_name: oldFormFields.playlist_name,
+        description: oldFormFields.description
+      }))
+    }
+  }, [metadata])
+
+  const onDropArtwork = async (selectedFiles: any, source: any) => {
+    setErrors({
+      ...errors,
+      artwork: false
+    })
+    try {
+      let file = selectedFiles[0]
+      file = await resizeImage(file)
+      const url = URL.createObjectURL(file)
+      setFormFields((formFields: PlaylistFormFields) => ({
+        ...formFields,
+        artwork: { file, url, source }
+      }))
+    } catch (err) {
+      setFormFields((formFields: PlaylistFormFields) => ({
+        ...formFields,
+        artwork: {
+          ...(formFields.artwork || {}),
+          error: err instanceof Error ? err.message : 'Unknown error'
+        }
+      }))
+    }
+  }
+
+  const onChangePlaylistName = (name: string) => {
+    setFormFields((formFields: PlaylistFormFields) => ({
+      ...formFields,
+      playlist_name: name
+    }))
+    if (name) {
+      setErrors({ ...errors, playlistName: false })
+    }
+  }
+
+  const onChangeDescription = (description: string) => {
+    setFormFields((formFields: PlaylistFormFields) => ({
+      ...formFields,
+      description
+    }))
+  }
+
+  const onSave = () => {
+    if (!formFields.playlist_name || (!formFields.artwork.file && !coverArt)) {
+      setErrors({
+        ...errors,
+        artwork: !formFields.artwork.file && !coverArt,
+        playlistName: !formFields.playlist_name
+      })
+    } else {
+      onSaveParent(formFields)
+    }
+  }
+
+  return (
+    <div>
```

Comment:
do we want to make this a real <form /> ?

Reply:
did you mean to type form at the end here? haha

Reply:
Haha so note to self if you use jsx syntax and don't quote it, it just disappears :D, yes!

Reply:
haha gotcha. i think we should but maybe in a separate PR (this one can just aim for parity with the existing code). not sure if the `UploadArtwork` will play nicely with forms